NeuroID/src/main/java/com/neuroid/tracker/callbacks/NIDSensorHelper.kt

```diff
@@ -49,21 +55,31 @@ object NIDSensorHelper {
     // only called by unit tests, do not call outside of tests
     internal fun resetSensorManager() {
         sensorManager = null
+        sensorActive = false
     }
 
 
     fun stopSensors() {
         sensorManager?.unregisterListener(listener)
+        sensorActive = false
     }
 
     fun restartSensors(nSensors: NIDSensors = nidSensors) {
+        if (sensorActive) {
+            return
+        }
+        // need to unregister the listeners before restarting the sensors.
+        stopSensors()
+
         listener = getNIDGenListener(nSensors, firstValuesGyro, firstValuesAccel)
```

Comment:
I agree, I think it might be because they wanted to ensure `listener` was not null when used in the following line to register the listener. Because technically you could call `restartSensors` without calling the `initSensorHelper` function and it would cause an error. I think we can put this in another ticket to refactor into a class with a proper initializer and then remove this